content/docs/guides/benchmarking-latency.md

```diff
@@ -0,0 +1,122 @@
+
+- **Cold queries**: Occur when a previously suspended compute resource is activated to process a request. This activation typically adds a few hundred milliseconds of latency. Cold queries are common in development or test environments where databases aren't running continuously.
+
+- **Hot queries**: Execute on an already-active database instance, delivering consistent low latency. These represent typical performance in production environments where databases run continuously or remain active most of the time.
+
+Free-tier Neon databases automatically suspend after 5 minutes of inactivity. Paid plans allow you to configure or disable the auto-suspend timeout, enabling you to customize your testing approach or eliminate cold starts entirely. See [Compute Lifecycle](/docs/introduction/compute-lifecycle) and [Auto-suspend Configuration](/docs/introduction/auto-suspend) for more details.
+
+## Benchmarking methodology
+
+For accurate benchmarking, always measure cold and hot queries separately:
+
+1. **Cold query testing**:
+
+   - Ensure your database is in a suspended state
+   - Make a request to trigger compute activation
+   - Measure this connection time, which includes the startup overhead
+
+2. **Hot query testing**:
+   - After triggering compute activation with a cold query
+   - Make subsequent requests within the active window
+   - Measure these connection times, which reflect normal operation
+
+This methodology isolates the cold start overhead from normal operating performance, giving you a clearer picture of both typical performance and worst-case latency.
+
+## Testing environment considerations
+
+Before running benchmarks, determine exactly what kind of latency you want to measure:
+
+- **Server-to-database latency**: If you're testing how quickly your application server can communicate with Neon, run benchmarks from the same location as your server. This is typically the most relevant metric for API performance.
+
+- **Client-to-database latency**: If you're testing direct client connections (rare in production), benchmark from client locations.
+
+Once you've determined what you're measuring:
+
+- **Test from your production region**: Geographic proximity is the primary factor in connection latency. Run benchmarks from the same region as your production environment to get accurate results. If your Neon database is in `us-east-1`, execute benchmarks from a server in that AWS region.
+
+- **Avoid localhost testing**: Testing from your local workstation doesn't reflect real-world conditions. In production, databases are typically queried from deployed servers, not client machines.
+
+Avoid testing across unrealistic distances that don't represent your production setup, as this introduces network overhead your users won't experience. For more on geographic factors affecting latency, see [Connection Latency and Timeouts](/docs/connect/connection-latency).
+
+## Connection types and their impact
```

Comment:
Maybe highlight that these connection types are for serverless environments where long-running traditional TCP Postgres connections aren't available or would add too much of an overhead as they would need be established for every incoming request.

Reply:
Done!